File: Examples/VisualBasic/AsposePDF/Security-Signatures/ExtractingImage.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports Aspose.Pdf.Forms
Imports System.Drawing

Namespace AsposePDF.SecuritySignatures
    Public Class ExtractingImage
        Public Shared Sub Run()
            ' ExStart:ExtractingImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_SecuritySignatures()

            Dim input As String = dataDir & Convert.ToString("ExtractingImage.pdf")
            Using pdfDocument As New Document(input)
                For Each field As Field In pdfDocument.Form
                    Dim sf As SignatureField = TryCast(field, SignatureField)
                    If sf IsNot Nothing Then
                        Dim outFile As String = dataDir & Convert.ToString("output_out_.jpg")
                        Using imageStream As Stream = sf.ExtractImage()
                            If imageStream IsNot Nothing Then
                                Using image As System.Drawing.Image = Bitmap.FromStream(imageStream)
                                    image.Save(outFile, System.Drawing.Imaging.ImageFormat.Jpeg)
                                End Using
                            End If
                        End Using
                    End If
                Next
            End Using
            ' ExEnd:ExtractingImage
        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Security-Signatures/ExtractingImage.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.InteractiveFeatures.Forms;
using System.Drawing;

namespace CSharp.AsposePdf.SecuritySignatures
{
    public class ExtractingImage
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();

            string input = dataDir+ @"ExtractingImage.pdf";
            using (Document pdfDocument = new Document(input))
            {
                foreach (Field field in pdfDocument.Form)
                {
                    SignatureField sf = field as SignatureField;
                    if (sf != null)
                    {
                        string outFile = dataDir+ @"output.jpg";
                        using (Stream imageStream = sf.ExtractImage())
                        {
                            if (imageStream != null)
                            {
                                using (System.Drawing.Image image = Bitmap.FromStream(imageStream))
                                {
                                    image.Save(outFile, System.Drawing.Imaging.ImageFormat.Jpeg);
                                }
                            }
                        }
                    }
                }
            }
            
            
        }
    }
}